docs/react/3.optimization/error-boundaries.md

```diff
@@ -0,0 +1,158 @@
+---
+title: Error Boundaries
+sidebar_label: Error Boundaries
+---
+
+import Img from '../../../src/components/Img';
+
+In the past, JavaScript errors inside components used to corrupt React’s internal state and cause it to emit cryptic errors on next renders. These errors were always caused by an earlier error in the application code, but React did not provide a way to handle them gracefully in components, and could not recover from them.
+
+## Introducing Error Boundaries
+
+A JavaScript error in a part of the UI shouldn’t break the whole app. To solve this problem for React users, React 16 introduces a new concept of an “error boundary”.
+
+Error boundaries are React components that **catch JavaScript errors anywhere in their child component tree, log those errors, and display a fallback UI** instead of the component tree that crashed. Error boundaries catch errors during rendering, in lifecycle methods, and in constructors of the whole tree below them.
```

Comment:
`in lifecycle methods, and in constructors of the whole tree below them.` could be highlighted. I think it's more important.